noir/noir-repo/acvm-repo/acir/src/circuit/mod.rs

```diff
@@ -59,34 +59,33 @@ pub struct Circuit {
     pub public_parameters: PublicInputs,
     /// The set of public inputs calculated within the circuit.
     pub return_values: PublicInputs,
-    /// Maps opcode locations to failed assertion messages.
-    /// These messages are embedded in the circuit to provide useful feedback to users
+    /// Maps opcode locations to failed assertion payloads.
+    /// The data in the payload is embedded in the circuit to provide useful feedback to users
     /// when a constraint in the circuit is not satisfied.
     ///
     // Note: This should be a BTreeMap, but serde-reflect is creating invalid
     // c++ code at the moment when it is, due to OpcodeLocation needing a comparison
     // implementation which is never generated.
-    //
-    // TODO: These are only used for constraints that are explicitly created during code generation (such as index out of bounds on slices)
-    // TODO: We should move towards having all the checks being evaluated in the same manner
-    // TODO: as runtime assert messages specified by the user. This will also be a breaking change as the `Circuit` structure will change.
-    pub assert_messages: Vec<(OpcodeLocation, String)>,
+    pub assert_messages: Vec<(OpcodeLocation, AssertionPayload)>,
 
     /// States whether the backend should use a SNARK recursion friendly prover.
     /// If implemented by a backend, this means that proofs generated with this circuit
     /// will be friendly for recursively verifying inside of another SNARK.
     pub recursive: bool,
 }
 
-impl Circuit {
-    /// Returns the assert message associated with the provided [`OpcodeLocation`].
-    /// Returns `None` if no such assert message exists.
-    pub fn get_assert_message(&self, opcode_location: OpcodeLocation) -> Option<&str> {
-        self.assert_messages
-            .iter()
-            .find(|(loc, _)| *loc == opcode_location)
-            .map(|(_, message)| message.as_str())
-    }
+pub const STRING_ERROR_SELECTOR: u64 = 0;
```

Comment:
It would be good to have some documentation here. We don't need a selector for static strings so it begs the question on why we need a selector.

Reply:
Added a doc comment!